tests/main.test.ts

```diff
@@ -480,6 +486,16 @@ it(`should support disabling the network accesses from the environment`, async (
 });
 
 describe(`read-only and offline environment`, () => {
+  let home: PortablePath;
+  beforeEach(() => {
+    home = npath.toPortablePath(process.env.COREPACK_HOME!);
+  });
+
+  afterEach(async () => {
+    await xfs.chmodPromise(ppath.join(home, `lastKnownGood.json`), 0o644);
+    await xfs.chmodPromise(home, 0o755);
```

Comment:
Could we add it only on the test that needs it? You can use `t.after` I think

Reply:
`onTestFinished` does not work because the cleanups from all `beforeEach` block are run before the hooks registered with `onTestFinished`. So the cleanup tries to remove `home` before the permissions are reverted and removal throws an exception.

The cleanup code is scoped to a `describe` block and required by all tests in that block so it doesn’t run unneeded.